android/api.md

```diff
@@ -566,6 +566,278 @@ private static String MyOIDCAuthProvider(){
+
+                    //Trigger the view to refresh by calling notifyDataSetChanged method
+                    notifyDataSetChanged();
+                }
+            });
+        }
+
+        @Override
+        public void onFailure(@Nonnull ApolloException e) {
+            Log.e(TAG, "Error " + e.getLocalizedMessage());
+        }
+        
+        @Override
+        public void onCompleted() {
+            Log.d(TAG, "Received onCompleted on subscription");
+        }
+    };
+``` 
+
+The `updateListPostsQueryCache` method will update the query cache of the `listPosts` query as shown below.
+
+```java
+    private void updateListPostsQueryCache() {
+        List<ListPostsQuery.ListPost> items = new ArrayList<>();
+        items.addAll(allPosts.values());
+
+        ListPostsQuery.Data data = new ListPostsQuery.Data(items);
+        ClientFactory.getInstance(display.getApplicationContext()).getStore()
+                .write(listPostsQuery, data).enqueue(null);
+    }
+```
+
+**Delta Query and Delta Query Callback**
+Next we will setup the delta query. We will use the `listPostsDelta` query for the changes made to Posts. The SDK will keep track of when it was last run and will automatically pass in the appropriate value for `lastSync` to get the updates in an optimized manner.
+
+```java
+    //Setup the Delta Query to get changes to posts incrementally
+    Query listPostsDeltaQuery = ListPostsDeltaQuery.builder().build();
+
+```
+
+In the `listPostsDeltaQueryCallback`, we will add, update, delete the changed Posts in the allPosts Map object and trigger the RecyclerView to refresh the contents of the view. We will also add the logic to update the query cache for the `listPosts` query.
+
+```java
+GraphQLCall.Callback listPostsDeltaQueryCallback = new GraphQLCall.Callback<ListPostsDeltaQuery.Data>() {
+            @Override
+            public void onResponse(@Nonnull final Response<ListPostsDeltaQuery.Data> response) {
+                if (response == null  || response.data() == null || response.data().listPostsDelta() == null ) {
+                    Log.d(TAG, "listPostsDelta returned with no data");
+                    return;
+                }
+
+                //Add to the ListPostsQuery Cache
+                Log.d(TAG, "listPostsDelta returned data. Iterating...");
+                display.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        for (ListPostsDeltaQuery.ListPostsDeltum p: response.data().listPostsDelta() ) {
+                            // Delete the Post if the aws_ds has the DELETE tag
+                            if ( DeltaAction.DELETE == p.aws_ds()) {
+                                mPostIDs.remove(p.id());
+                                allPosts.remove(p.id());
+                                continue;
+                            }
+
+                            // Add or Update the post otherwise
```

Comment:
Delete these 3 lines 

// Add or Update the post otherwise
                            if (allPosts.get(p.id()) == null ) {
                                mPostIDs.add(0, p.id());
                            }